contracts/base/governance/ATMToken.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.5.17;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20Detailed.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20Mintable.sol";
+import "@openzeppelin/contracts/lifecycle/Pausable.sol";
+
+/**
+  *  @title ATM Token for Teller DAO
+  *
+  *  @author develop@teller.finance
+ */
+
+contract ATMToken is ERC20Detailed, Pausable, ERC20Mintable {
+    /**
+      *  @dev ATMToken implements an ERC20 token with a supply cap and a vesting scheduling
+     */
+
+    // State Variables
+    uint256 private _cap;
+
+    // Struct
+    struct VestingTokens {
+        address beneficiary;
+        uint256 amount;
+        uint256 deadline;
+    }
+
+    // Mappings
+    mapping (address => VestingTokens) private _vestingBalances;
+
+    // Events
+    event NewCap(uint256 newcap);
```

Comment:
Please, create an interface with events and functions.

Reply:
Created